Sources/Network/NetworkAuthenticator.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  Authenticator.swift
+//  Alicerce
+//
+//  Created by Luís Portela on 03/07/2017.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import Foundation
+
+public protocol NetworkAuthenticator {
+    func authenticate(request: URLRequest, _ completion: @escaping (URLRequest) -> Void)
+
+    func shouldRetry(with data: Data?, response: HTTPURLResponse?, error: Swift.Error?) -> Bool
```

Comment:
Did I say something wrong? I don't think this attitude is constructive. 

If you *read* what I wrote, you will be able to find the exact argument I'm using. 

You think it's unnecessary complexity, fine, just say so. Geez